consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validation/RoundChangeMessageValidator.java

```diff
@@ -14,11 +14,133 @@
  */
 package org.hyperledger.besu.consensus.qbft.validation;
 
+import static org.hyperledger.besu.consensus.qbft.validation.ValidationHelpers.hasDuplicateAuthors;
+import static org.hyperledger.besu.consensus.qbft.validation.ValidationHelpers.hasSufficientEntries;
+
+import org.hyperledger.besu.consensus.common.bft.BftExtraData;
+import org.hyperledger.besu.consensus.common.bft.ConsensusRoundIdentifier;
+import org.hyperledger.besu.consensus.common.bft.payload.SignedData;
 import org.hyperledger.besu.consensus.qbft.messagewrappers.RoundChange;
+import org.hyperledger.besu.consensus.qbft.payload.PreparePayload;
+import org.hyperledger.besu.consensus.qbft.payload.PreparedRoundMetadata;
+import org.hyperledger.besu.ethereum.BlockValidator;
+import org.hyperledger.besu.ethereum.BlockValidator.BlockProcessingOutputs;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.mainnet.HeaderValidationMode;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class RoundChangeMessageValidator {
 
-  public boolean validateRoundChange(final RoundChange msg) {
+  private static final String ERROR_PREFIX = "Invalid RoundChange Message";
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final RoundChangePayloadValidator roundChangePayloadValidator;
+  private final long quorumMessageCount;
+  private final long chainHeight;
+  private final Collection<Address> validators;
+  private final BlockValidator blockValidator;
+  private final ProtocolContext protocolContext;
+
+  public RoundChangeMessageValidator(
+      final RoundChangePayloadValidator roundChangePayloadValidator,
+      final long quorumMessageCount,
+      final long chainHeight,
+      final Collection<Address> validators,
+      final BlockValidator blockValidator,
+      final ProtocolContext protocolContext) {
+    this.roundChangePayloadValidator = roundChangePayloadValidator;
+    this.quorumMessageCount = quorumMessageCount;
+    this.chainHeight = chainHeight;
+    this.validators = validators;
+    this.blockValidator = blockValidator;
+    this.protocolContext = protocolContext;
+  }
+
+  public boolean validate(final RoundChange msg) {
+
+    if (!roundChangePayloadValidator.validate(msg.getSignedPayload())) {
```

Comment:
height should be checked that it matches as well

Reply:
This check exists in the RoundChangePayloadValidator.